addons/web/controllers/profiling.py

```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+import json
+
+from odoo.exceptions import UserError
+from odoo.http import Controller, request, Response, route
+
+class Profiling(Controller):
+
+    @route('/web/set_profiling', type='http', auth='public', sitemap=False)
```

Comment:
why `public`? shouldn't this be restricted to admin?

Reply:
No it is exactly there to allow to profile as a public user when rpc is not available. This route should return an error if profiling is disabled on this database.